environment_all_backends.yml

```diff
@@ -16,11 +16,12 @@ dependencies:
   # Numerical backend installations with CUDA support where possible:
   - numpy>=1.19.5
   - cudatoolkit>=11.1
-  - pytorch>=1.9 # CPU version
-  - tensorflow>=2.10.0 # CPU version
+  - pytorch>=1.9=*cuda*
+  - tensorflow-gpu>=2.6.0
   # jaxlib with CUDA support is not available for conda
   - pip:
+      - --find-links https://storage.googleapis.com/jax-releases/jax_cuda_releases.html
```

Comment:
@ilan-gold Ah, this is likely the problem btw! these change break the CI. You need the same env as currently on main
https://github.com/esa/torchquad/blob/main/environment_all_backends.yml

Same for #141

Reply:
Thanks! Sorry was a little busy yesterday finding small bugs: https://github.com/esa/torchquad/pull/141#issuecomment-1396809918

Reply:
dealing with this now!

Reply:
```
absl-py                       1.4.0
aiohttp                       3.8.3
aiosignal                     1.3.1
alabaster                     0.7.13
appdirs                       1.4.4
astunparse                    1.6.3
async-timeout                 4.0.2
attrs                         22.2.0
autoray                       0.0.0
Babel                         2.11.0
blinker                       1.5
brotlipy                      0.7.0
cached-property               1.5.2
cachetools                    5.2.1
certifi                       2022.12.7
cffi                          1.15.1
charset-normalizer            2.1.1
click                         8.1.3
colorama                      0.4.6
contourpy                     1.0.7
cryptography                  39.0.0
cycler                        0.11.0
docutils                      0.19
exceptiongroup                1.1.0
flatbuffers                   23.1.4
fonttools                     4.38.0
frozenlist                    1.3.3
gast                          0.4.0
google-auth                   2.16.0
google-auth-oauthlib          0.4.6
google-pasta                  0.2.0
grpcio                        1.51.1
h5py                          3.7.0
idna                          3.4
imagesize                     1.4.1
importlib-metadata            6.0.0
iniconfig                     2.0.0
jax                           0.4.1
jaxlib                        0.4.1
Jinja2                        3.1.2
keras                         2.11.0
Keras-Preprocessing           1.1.2
kiwisolver                    1.4.4
loguru                        0.6.0
Markdown                      3.4.1
MarkupSafe                    2.1.1
matplotlib                    3.6.3
multidict                     6.0.4
munkres                       1.1.4
numpy                         1.24.1
oauthlib                      3.2.2
opt-einsum                    3.3.0
packaging                     23.0
Pillow                        9.4.0
pip                           22.3.1
pluggy                        1.0.0
ply                           3.11
pooch                         1.6.0
protobuf                      4.21.12
pyasn1                        0.4.8
pyasn1-modules                0.2.7
pycparser                     2.21
Pygments                      2.14.0
PyJWT                         2.6.0
pyOpenSSL                     23.0.0
pyparsing                     3.0.9
PyQt5                         5.15.7
PyQt5-sip                     12.11.0
PySocks                       1.7.1
pytest                        7.2.1
python-dateutil               2.8.2
pytz                          2022.7.1
pyu2f                         0.1.5
requests                      2.28.2
requests-oauthlib             1.3.1
rsa                           4.9
scipy                         1.10.0
setuptools                    66.0.0
sip                           6.7.5
six                           1.16.0
snowballstemmer               2.2.0
Sphinx                        6.1.3
sphinx-rtd-theme              0.5.2
sphinxcontrib-applehelp       1.0.2
sphinxcontrib-devhelp         1.0.2
sphinxcontrib-htmlhelp        2.0.0
sphinxcontrib-jsmath          1.0.1
sphinxcontrib-qthelp          1.0.3
sphinxcontrib-serializinghtml 1.1.5
tensorboard                   2.11.2
tensorboard-data-server       0.6.1
tensorboard-plugin-wit        1.8.1
tensorflow                    2.11.0
tensorflow-estimator          2.11.0
termcolor                     2.2.0
toml                          0.10.2
tomli                         2.0.1
torch                         1.13.1.post200
tornado                       6.2
tqdm                          4.64.1
typing_extensions             4.4.0
unicodedata2                  15.0.0
urllib3                       1.26.14
Werkzeug                      2.2.2
wheel                         0.38.4
wrapt                         1.14.1
yarl                          1.8.2
zipp                          3.11.0
```
This is my pip list.  The only thing that appears to have been broken is one of the MC tests. I am using python3.10 which could be a problem but there is no claim about version in the repo.  Can you reproduce what is happening on CI locally?  It looks like an issue with the tensorflow import.  In any case, we should then probably pin a version in the `conda` file maybe?

Reply:
And furthermore, that test did not break in python3.9.  I think it's probably best to expand testing to 3.8-10 (or 11) and make sure everything works, but I also do not know enough about the project to make that call.  Happy to take your order here.

Reply:
`conda` also does indeed seem to find "conflicts" with 3.9 which maybe is why things were taking a while for you.  none were found with 3.10.

Reply:
> ```
> absl-py                       1.4.0
> aiohttp                       3.8.3
> aiosignal                     1.3.1
> alabaster                     0.7.13
> appdirs                       1.4.4
> astunparse                    1.6.3
> async-timeout                 4.0.2
> attrs                         22.2.0
> autoray                       0.0.0
> Babel                         2.11.0
> blinker                       1.5
> brotlipy                      0.7.0
> cached-property               1.5.2
> cachetools                    5.2.1
> certifi                       2022.12.7
> cffi                          1.15.1
> charset-normalizer            2.1.1
> click                         8.1.3
> colorama                      0.4.6
> contourpy                     1.0.7
> cryptography                  39.0.0
> cycler                        0.11.0
> docutils                      0.19
> exceptiongroup                1.1.0
> flatbuffers                   23.1.4
> fonttools                     4.38.0
> frozenlist                    1.3.3
> gast                          0.4.0
> google-auth                   2.16.0
> google-auth-oauthlib          0.4.6
> google-pasta                  0.2.0
> grpcio                        1.51.1
> h5py                          3.7.0
> idna                          3.4
> imagesize                     1.4.1
> importlib-metadata            6.0.0
> iniconfig                     2.0.0
> jax                           0.4.1
> jaxlib                        0.4.1
> Jinja2                        3.1.2
> keras                         2.11.0
> Keras-Preprocessing           1.1.2
> kiwisolver                    1.4.4
> loguru                        0.6.0
> Markdown                      3.4.1
> MarkupSafe                    2.1.1
> matplotlib                    3.6.3
> multidict                     6.0.4
> munkres                       1.1.4
> numpy                         1.24.1
> oauthlib                      3.2.2
> opt-einsum                    3.3.0
> packaging                     23.0
> Pillow                        9.4.0
> pip                           22.3.1
> pluggy                        1.0.0
> ply                           3.11
> pooch                         1.6.0
> protobuf                      4.21.12
> pyasn1                        0.4.8
> pyasn1-modules                0.2.7
> pycparser                     2.21
> Pygments                      2.14.0
> PyJWT                         2.6.0
> pyOpenSSL                     23.0.0
> pyparsing                     3.0.9
> PyQt5                         5.15.7
> PyQt5-sip                     12.11.0
> PySocks                       1.7.1
> pytest                        7.2.1
> python-dateutil               2.8.2
> pytz                          2022.7.1
> pyu2f                         0.1.5
> requests                      2.28.2
> requests-oauthlib             1.3.1
> rsa                           4.9
> scipy                         1.10.0
> setuptools                    66.0.0
> sip                           6.7.5
> six                           1.16.0
> snowballstemmer               2.2.0
> Sphinx                        6.1.3
> sphinx-rtd-theme              0.5.2
> sphinxcontrib-applehelp       1.0.2
> sphinxcontrib-devhelp         1.0.2
> sphinxcontrib-htmlhelp        2.0.0
> sphinxcontrib-jsmath          1.0.1
> sphinxcontrib-qthelp          1.0.3
> sphinxcontrib-serializinghtml 1.1.5
> tensorboard                   2.11.2
> tensorboard-data-server       0.6.1
> tensorboard-plugin-wit        1.8.1
> tensorflow                    2.11.0
> tensorflow-estimator          2.11.0
> termcolor                     2.2.0
> toml                          0.10.2
> tomli                         2.0.1
> torch                         1.13.1.post200
> tornado                       6.2
> tqdm                          4.64.1
> typing_extensions             4.4.0
> unicodedata2                  15.0.0
> urllib3                       1.26.14
> Werkzeug                      2.2.2
> wheel                         0.38.4
> wrapt                         1.14.1
> yarl                          1.8.2
> zipp                          3.11.0
> ```
> 
> This is my pip list. 

For a full list you need to use `conda list` I think. 

> Can you reproduce what is happening on CI locally?

Yes, should be possible with https://github.com/nektos/act I think?

I might be wrong but my suspicion would be it is the same problem as here https://github.com/esa/torchquad/pull/158#issuecomment-1369729268

Reply:
> And furthermore, that MC test did not break in python3.9 (at least on the CI). I think it's probably best to expand testing to 3.8-10 (or 11) and make sure everything works, but I also do not know enough about the project to make that call. Happy to take your order here.

You mean setting up the CI to run with different python version? We can do that sure, it's basically just modifying [this](https://github.com/esa/torchquad/blob/main/.github/workflows/run_tests.yml) to run it with several python version. We can create a small separate PR to quickly merge into main and develop for that if you want?

Reply:
Ok I think I need to install `mamba` and then proceed from there then.